lib/Models/Terria.ts

```diff
@@ -1053,6 +1063,38 @@ export default class Terria {
     this.loadPersistedMapSettings();
   }
 
+  /**
+   * Zoom to workbench items if `focusWorkbenchItemsAfterLoadingInitSources` is `true`.
+   *
+   * Note that the current behaviour is to zoom to the first item of the
+   * workbench, however in the future we should modify it to zoom to a view
+   * which shows all the workbench items.
+   *
+   * If a Cesium or Leaflet viewer is not available,
+   * we wait for it to load before triggering the zoom.
+   */
+  private async doZoomToWorkbenchItems() {
+    if (!this.focusWorkbenchItemsAfterLoadingInitSources) {
+      return;
+    }
+
+    // TODO: modify this to zoom to a view that shows all workbench items
+    // instead of just zooming to the first workbench item!
+    const firstMappableItem = this.workbench.items.find((item) =>
+      MappableMixin.isMixedInto(item)
+    ) as MappableMixin.Instance | undefined;
+    if (firstMappableItem) {
+      // When the app loads, Cesium/Leaflet viewers are loaded
+      // asynchronously. Until they become available, a stub viewer called
+      // `NoViewer` is used. `NoViewer` does not implement zooming to mappable
+      // items. So here wait for a valid viewer to become available before
+      // attempting to zoom to the mappable item.
+      const isViewerAvailable = () => this.currentViewer.type !== NoViewer.type;
+      await when(isViewerAvailable);
+      await this.currentViewer.zoomTo(firstMappableItem, 0.0);
```

Comment:
Not sure I understand why that check is required?

If a viewer is currently loading, then `when` will re-run the `isViewerAvailable` test whenever `currentViewer` is updated. Or is there something else I'm missing?

Reply:
If `mainViewer.viewerMode` is undefined, then this.currentViewer.type will be permanently NoViewer and we'll have a hanging promise.

I think you could wait for `mainViewer.viewerLoadPromise`, check that the viewer isn't NoViewer afterwards then run zoomTo

Reply:
@steve9164 - as discussed offline, I'm going with the option of keeping this code and leaving a note on hanging promise. 

Discussed reason: Because this piece of code runs before react renders, `viewerLoadPromise` will always resolve successfully to a `NoViewer` instance. So the suggested workaround will fail.